### PR TITLE
Feature/interceptor for activities

### PR DESCRIPTION
**Pull request by Gelphy**
Add first interceptor for roles and delete unneeded tests and controllers
Fix assign role attribute
Add extensions and constants, refactor attribute
Add extensions, refactor attribute
Remove GetCollaborators method from extensions
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63165013%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63165174%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63183902%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63185682%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63195587%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63195814%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63202669%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63202694%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205441%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205710%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205751%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63292491%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63292504%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63292509%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63296623%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63297419%22%2C%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63297431%22%5D%2C%20%22comments%22%3A%20%7B%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%2016%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205710%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20are%20using%20this%20method%20only%20in%20one%20place.%20Why%20it%20should%20be%20public%3F%20Incapsulate%20all%20unnessesary%20information%20that%20is%20not%20available%20for%20client.%22%2C%20%22created_at%22%3A%20%222016-05-13T15%3A47%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22Already%20done%22%2C%20%22created_at%22%3A%20%222016-05-15T14%3A29%3A21Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%3AL1-55%22%7D%2C%20%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs%208%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63165013%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Command%20Context%20is%20IDisposable%3F%20Indeed%3F%20Please%20describe%20case.%22%2C%20%22created_at%22%3A%20%222016-05-13T10%3A48%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22%20If%20I%20want%20to%20register%20type%20in%20OwinContext%20it%20has%20to%20implement%20IDisposable%22%2C%20%22created_at%22%3A%20%222016-05-13T13%3A51%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%2C%20%7B%22body%22%3A%20%22You%20really%20needed%20OWIN%20context%3F%20OWIN%20in%20our%20case%20it%27s%20like%20DI%20as%20SimpleInjector.%20I%20suggest%20that%20you%20can%20you%20use%20SimpleInjector%20instead.%22%2C%20%22created_at%22%3A%20%222016-05-13T14%3A47%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27ll%20fix%20it%22%2C%20%22created_at%22%3A%20%222016-05-13T15%3A30%3A00Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs%3AL1-10%22%7D%2C%20%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs%2033%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63195814%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Where%20interfaces%3F%20Why%20you%20are%20using%20concrete%20types%20ActivityContextCommand%2C%20ActivityContextQuery%20%3F%22%2C%20%22created_at%22%3A%20%222016-05-13T14%3A49%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27ll%20fix%20it%22%2C%20%22created_at%22%3A%20%222016-05-13T15%3A30%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs%3AL1-59%22%7D%2C%20%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205751%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Same.%22%2C%20%22created_at%22%3A%20%222016-05-13T15%3A47%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22already%20done%22%2C%20%22created_at%22%3A%20%222016-05-15T14%3A29%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%3AL1-55%22%7D%2C%20%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63165174%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Abstract%20class%20without%20any%20line%20of%20code%3F%22%2C%20%22created_at%22%3A%20%222016-05-13T10%3A50%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22will%20change%20to%20%20interface%20wich%20marked%20my%20attributes%5Cr%5Cnpublic%20interface%20IInternalActivityFilter%20%7B%20%7D%22%2C%20%22created_at%22%3A%20%222016-05-13T13%3A40%3A38Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs%3AL1-7%22%7D%2C%20%22Pull%200946fc118c21a86997c8ef17bd26b7a7023571a5%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%2013%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63205441%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Can%20you%20reso%3Bve%20all%20dependencies%20through%20the%20SimpleInjector%20intead%20of%20spreading%20resolving%20instances%20in%20different%20places%3F%20You%20you%20should%20have%20only%20one%20point%20of%20injection%20and%20resolving.%22%2C%20%22created_at%22%3A%20%222016-05-13T15%3A45%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22when%20injecting%20dependencies%20into%20Web%20API%20filter%20attributes%2C%20get%20my%20instances%20from%20dependency%20resolver%20http%3A//simpleinjector.readthedocs.io/en/latest/webapiintegration.html%20%2C%20according%20to%20this%20documentation.%20For%20example%3A%5Cr%5Cnvar%20checker%20%3D%20GlobalConfiguration.Configuration.DependencyResolver%5Cr%5Cn%20%20%20%20%20%20%20%20%20%20%20%20.GetService%28typeof%28IMinimumAgeChecker%29%29%20as%20IMinimumAgeChecker%3B%22%2C%20%22created_at%22%3A%20%222016-05-15T14%3A28%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs%3AL1-55%22%7D%2C%20%22Pull%20ca20da879419d370eb8c1f3a36740804b5020a9c%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs%207%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/ITsvetkoFF/Kv-013/pull/152%23discussion_r63296623%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Is%20it%20correct%20name%20for%20class%20holding%20extension%20methods%20for%20IPrincipal%20interface%3F%22%2C%20%22created_at%22%3A%20%222016-05-15T19%3A24%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6068068%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/TerraVenil%22%7D%7D%2C%20%7B%22body%22%3A%20%22No.%20Correct%20name%20is%20IdentityExtensions%22%2C%20%22created_at%22%3A%20%222016-05-15T20%3A22%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%27ll%20fix%20it%22%2C%20%22created_at%22%3A%20%222016-05-15T20%3A23%3A36Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/10926945%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Gelphy%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs%3AL1-25%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs 8'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63165013'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs:L1-10</a></b>
- <a href='https://github.com/TerraVenil'></a> Command Context is IDisposable? Indeed? Please describe case.
- <a href='https://github.com/Gelphy'></a> If I want to register type in OwinContext it has to implement IDisposable
- <a href='https://github.com/TerraVenil'></a> You really needed OWIN context? OWIN in our case it's like DI as SimpleInjector. I suggest that you can you use SimpleInjector instead.
- <a href='https://github.com/Gelphy'></a> I'll fix it
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs 5'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63165174'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs:L1-7</a></b>
- <a href='https://github.com/TerraVenil'></a> Abstract class without any line of code?
- <a href='https://github.com/Gelphy'></a> will change to  interface wich marked my attributes
  public interface IInternalActivityFilter { }
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs 33'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63195814'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs:L1-59</a></b>
- <a href='https://github.com/TerraVenil'></a> Where interfaces? Why you are using concrete types ActivityContextCommand, ActivityContextQuery ?
- <a href='https://github.com/Gelphy'></a> I'll fix it
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs 13'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63205441'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs:L1-55</a></b>
- <a href='https://github.com/TerraVenil'></a> Can you reso;ve all dependencies through the SimpleInjector intead of spreading resolving instances in different places? You you should have only one point of injection and resolving.
- <a href='https://github.com/Gelphy'></a> when injecting dependencies into Web API filter attributes, get my instances from dependency resolver http://simpleinjector.readthedocs.io/en/latest/webapiintegration.html , according to this documentation. For example:
  var checker = GlobalConfiguration.Configuration.DependencyResolver
  .GetService(typeof(IMinimumAgeChecker)) as IMinimumAgeChecker;
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs 16'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63205710'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs:L1-55</a></b>
- <a href='https://github.com/TerraVenil'></a> You are using this method only in one place. Why it should be public? Incapsulate all unnessesary information that is not available for client.
- <a href='https://github.com/Gelphy'></a> Already done
- [ ] <a href='#crh-comment-Pull 0946fc118c21a86997c8ef17bd26b7a7023571a5 DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs 21'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63205751'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs:L1-55</a></b>
- <a href='https://github.com/TerraVenil'></a> Same.
- <a href='https://github.com/Gelphy'></a> already done
- [ ] <a href='#crh-comment-Pull ca20da879419d370eb8c1f3a36740804b5020a9c DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs 7'></a> &nbsp;<b><a href='https://github.com/ITsvetkoFF/Kv-013/pull/152#discussion_r63296623'>File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs:L1-25</a></b>
- <a href='https://github.com/TerraVenil'></a> Is it correct name for class holding extension methods for IPrincipal interface?
- <a href='https://github.com/Gelphy'></a> No. Correct name is IdentityExtensions
- <a href='https://github.com/Gelphy'></a> I'll fix it

<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/152?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/ITsvetkoFF/Kv-013/pull/152?approve=1'></a>&nbsp;<a href='https://github.com/ITsvetkoFF/Kv-013/pull/152'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs [hunk 1]
```diff
@@ -1,8 +1,9 @@
-﻿using GitHubExtension.Activity.DAL;
+﻿using System;
+using GitHubExtension.Activity.DAL;
 
 namespace GitHubExtension.Activity.Internal.WebApi.Commands
 {
-    public class ActivityContextCommand : IActivityContextCommand
+    public class ActivityContextCommand : IActivityContextCommand, IDisposable
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs]: Command Context is IDisposable? Indeed? Please describe case.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿using System.Web.Http.Filters;
+
+namespace GitHubExtension.Infrastructure.ActionFilters.InternalActivitiesFilters
+{
+    public abstract class InternalActivityFilter : ActionFilterAttribute { }
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs]: Abstract class without any line of code?

Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/InternalActivityFilter.cs]: will change to  interface wich marked my attributes
public interface IInternalActivityFilter { }

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs (hunk 1 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs]: If I want to register type in OwinContext it has to implement IDisposable

Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs]: You really needed OWIN context? OWIN in our case it's like DI as SimpleInjector. I suggest that you can you use SimpleInjector instead.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            User appUser = applicationUserManager.FindByGitHubId(gitHubId);
+            string collaboratorName = appUser.UserName;
+            var activityContextQuery = dependencyResolver.GetService<ActivityContextQuery>();
+            var activityContextCommand = dependencyResolver.GetService<ActivityContextCommand>();
+            var user = actionExecutedContext.GetUserModel();
+
+            AddRoleActivity(activityContextQuery, activityContextCommand, user, repoId, roleToAssign, collaboratorName); 
+        }
+
+        public void AddRoleActivity(ActivityContextQuery activityContextQuery,
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs]: Where interfaces? Why you are using concrete types ActivityContextCommand, ActivityContextQuery ?

File: DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs (hunk 1 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Activity.Internal.WebApi/Commands/ActivityContextCommand.cs]: I'll fix it

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs (hunk 3 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/InternalActivitiesFilters/AssignRoleToUserActivityAttribute.cs]: I'll fix it

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+using GitHubExtension.Infrastructure.ActionFilters.Models;
+using Microsoft.AspNet.Identity;
+
+namespace GitHubExtension.Infrastructure.ActionFilters.Extensions
+{
+    public static class ActivityFiltersExtensions
+    {
+        public static IDependencyResolver GetDependencyResolver(this HttpActionExecutedContext actionExecutedContext)
+        {
+            return actionExecutedContext.ActionContext.RequestContext.Configuration.DependencyResolver;
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: Can you reso;ve all dependencies through the SimpleInjector intead of spreading resolving instances in different places? You you should have only one point of injection and resolving.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        public static string GetUserId(this HttpActionExecutedContext actionExecutedContext)
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: You are using this method only in one place. Why it should be public? Incapsulate all unnessesary information that is not available for client.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            return actionExecutedContext.ActionContext.RequestContext.Principal.Identity.GetUserId();
+        }
+
+        public static string GetUserName(this HttpActionExecutedContext actionExecutedContext)
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: Same.

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs (hunk 4 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: when injecting dependencies into Web API filter attributes, get my instances from dependency resolver http://simpleinjector.readthedocs.io/en/latest/webapiintegration.html , according to this documentation. For example:
var checker = GlobalConfiguration.Configuration.DependencyResolver
            .GetService(typeof(IMinimumAgeChecker)) as IMinimumAgeChecker;

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs (hunk 5 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: Already done

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs (hunk 6 above)
Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.ActionFilters/Extensions/ActivityFiltersExtensions.cs]: already done

File: DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Security.Claims;
+using System.Security.Principal;
+using Microsoft.AspNet.Identity;
+
+namespace GitHubExtension.Infrastructure.Extensions.Identity
+{
+    public static class CurrentProjectExtensions
```
Comment (TerraVenil) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs]: Is it correct name for class holding extension methods for IPrincipal interface?

Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs]: No. Correct name is IdentityExtensions

Comment (Gelphy) [inline DevelopGitHub/GitHubExtension/GitHubExtension.Infrastructure.Extensions/Identity/CurrentProjectExtensions.cs]: I'll fix it

**Pull request merged**